pkg/client/helpers.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func Generate(ownershipLevel, action string, resourceHierarchy ...string) func(r *http.Request) string {
```

Comment:
In my experience with other stacks the `helpers` module often ends up as a dumping place for all kinds of unrelated logic. That said, and given that a Permission is an integral part of the IAM domain, I suggest us to move this logic out of there and place it in a `permissions.go` file.

We could also define a struct for a `permission` with the corresponding fields, and add this method as `buildPermission` that builds a struct. The stringification part could be left to a `String()` method, like:


```
pkg/client/permissions.go

type permission struct {
  ownershipLevel       string
  action                       string
  resourceHierarchy   string
}

func BuildPermission(ownershipLevel, action string, resourceHierarchy ...string ) (*permission, err) {
 ...
}

}

func (p permission) String() string {
  return fmt.Sprintf(
    ...
  )
}

```

WDYT?